### PR TITLE
sign with signify and a secret key designated at startup time

### PR DESCRIPTION
**Pull request by benadida**

### CODE REVIEW

**Review by umbernhard — APPROVED** [review]
This looks good to me, but I'll rely on @eventualbuddha's typescript knowledge. The usage of signify is correct.

File: src/ipc/sign.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+const debug = makeDebug('kiosk-browser:sign')
+
+export interface SignParams {
+  signatureType: string
+  payload: string
+}
+
+async function sign(
+  { signatureType, payload }: SignParams,
+  signifySecretKey: string | undefined,
```
Comment (eventualbuddha) [inline src/ipc/sign.ts:18]:
```suggestion
  signifySecretKey?: string,
```

File: src/ipc/sign.test.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  changedDevices,
+  autoconfiguredPrinter,
+  options: {
+    signifySecretKey: '/tmp/key.sec',
+    url: new URL('about:blank'),
+    originFilePermissions: [],
+  },
+})
+
+test('call to sign invoke the right signify command', async () => {
```
Comment (eventualbuddha) [inline src/ipc/sign.test.ts:30]: None of the error cases are tested. Add a couple?

File: src/ipc/sign.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  try {
+    const { stdout, stderr } = await exec(
+      'signify-openbsd',
+      ['-S', '-s', signifySecretKey, '-m', '-', '-x', '-'],
+      rawPayloadToSign,
+    )
+
+    if (stderr) {
+      debug('error trying to sign %s', stderr)
+      return undefined
```
Comment (eventualbuddha) [inline src/ipc/sign.ts:41]: Minor, but inconsistent about `return` vs `return undefined`.

**Review by eventualbuddha — APPROVED** [review]
There's an ESLint issue and this could use some tests for error cases, but otherwise LGTM.

**Pull request merged**